src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -785,24 +786,30 @@ public static Set<InetAddressAndPort> getPriorityHosts(RepairType repairType)
         return hosts;
     }
 
-    public static boolean checkNodeContainsKeyspaceReplica(Keyspace ks)
+    public static boolean shouldConsiderKeyspace(Keyspace ks)
     {
         AbstractReplicationStrategy replicationStrategy = ks.getReplicationStrategy();
-        boolean ksReplicaOnNode = true;
+        boolean repair = true;
         if (replicationStrategy instanceof NetworkTopologyStrategy)
         {
             Set<String> datacenters = ((NetworkTopologyStrategy) replicationStrategy).getDatacenters();
             String localDC = DatabaseDescriptor.getEndpointSnitch().getDatacenter(FBUtilities.getBroadcastAddressAndPort());
             if (!datacenters.contains(localDC))
             {
-                ksReplicaOnNode = false;
+                repair = false;
             }
         }
         if (replicationStrategy instanceof LocalStrategy)
         {
-            ksReplicaOnNode = false;
+            repair = false;
         }
-        return ksReplicaOnNode;
+        if (ks.getName().equalsIgnoreCase(SchemaConstants.TRACE_KEYSPACE_NAME))
```

Comment:
We could consider making this more robust in the future if we want to avoid repairing other keyspaces (like check against a list or something), but I think we can reason about that when we come to it, so I think this is good for now :+1:

Reply:
yup